File: sach/de3313.cpp
//DE3313.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
#define ll long long
bool doixung(ll m){
  string k,s;
  s = k = to_string(m);
  reverse(s.begin(), s.end());
  return s==k;
}
int main()
{
    ll N = 1e7;
    vector<bool> snt(N+1,true);
    snt[0] = snt[1] = false;
    for(int i=2;i*i<=N;i++){
      if(snt[i]==true){
        for(int j = i*i; j<=N;j+=i){
          snt[j] = false;
        }
      }
    }
    ll a,b,res = 0;
    cin>>a>>b;
    ll m=sqrt(a);
    if (m*m<a) m++;
    ll n=sqrt(b);
    for(ll i = m;i<=n;i++)
        if(snt[i] && doixung(i))
          res++;
    cout<<res;
    return 0;
}

File: sach/de3313.py
//DE3313.PY - Tran Huu Nam - hsgtin.vn - 2024
import math
a, b = [int(x) for x in input().split()]
a = math.ceil(math.sqrt(a))
b = math.floor(math.sqrt(b))
def snt(n):
    s = [True]*(n+1)
    s[0] = s[1] = False
    for i in range(1,int(math.sqrt(n))+1):
        if s[i] == True:
            for j in range(i*i, n+1, i):
                s[j] = False
    return s
s = snt(b)
dem = 0
for i in range(a, b+1):
    if s[i] and i == int(str(i)[::-1]):
        dem += 1
print(dem)

